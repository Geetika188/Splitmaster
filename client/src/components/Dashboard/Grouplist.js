import React from 'react';
import axios from 'axios';
import { serverIp, serverPort } from '../config';

class Grouplist extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      registeredStudents: [],
    };
    this.returnRegisteredStudents = this.returnRegisteredStudents.bind(this);
   
  }

  componentDidMount() {
    axios.post(`${serverIp}:${serverPort}/getStudentsRegisteredInAEvent`, { email: 'geetika.kapil@sjsu.edu' })
      .then((response) => {
        console.log('Response data in componentDidMount');
        console.log(response.data);
        this.setState({
          registeredStudents: response.data,
        });
      }).catch((err) => {
        console.log(`Error in componentDidMount of RegisteredStudents: ${err}`);
        window.alert('Error in connecting to server');
      });
  }

  
  returnRegisteredStudents() {
    return this.state.registeredStudents.map((eachStudent) => {
      let imgSrc = `${serverIp}:${serverPort}/default.png`;
      if (eachStudent.profile_picture_url !== '') {
        imgSrc = `${serverIp}:${serverPort}/${eachStudent.profile_picture_url}`;
      }
      return (
        <div>
            
               <span>{eachStudent.groupname} </span>
                <button >Detail</button>
              
        </div>
      );

    });
       
          
  }
  
  
render()
{
return(
<div>
<ul>
             <li className="friendlist">
               <i class="fas fa-user"/>
               
               {this.returnRegisteredStudents()}
               
               </li>


</ul>


</div>
);
}
}
export default Grouplist;
