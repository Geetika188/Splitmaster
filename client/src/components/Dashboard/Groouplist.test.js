import React from 'react';
import { mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import Grouplist from './Grouplist';

configure({ adapter: new Adapter() });

it('should render correctly', () => {
  const component = mount(<Grouplist/>);
  expect(component).toMatchSnapshot();
  component.unmount();
});
