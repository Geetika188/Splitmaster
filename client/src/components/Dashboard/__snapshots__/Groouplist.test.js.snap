// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Grouplist>
  <div>
    <ul>
      <li
        className="friendlist"
      >
        <i
          class="fas fa-user"
        />
      </li>
    </ul>
  </div>
</Grouplist>
`;
